pkg/services/ngalert/schedule/alert_rule.go

```diff
@@ -428,7 +425,7 @@ func (a *alertRule) evaluate(ctx context.Context, e *Evaluation, span trace.Span
 			err = results.Error()
 		}
 
-		logger.Debug("Alert rule evaluated", "error", err, "duration", dur)
+		logger.Error("Failed to evaluate rule", "error", err, "duration", dur)
```

Comment:
This is the same message as we have [here](https://github.com/grafana/grafana/blob/d999b415df78ceba5302e3c0f98975a870cce8c8/pkg/services/ngalert/schedule/alert_rule.go#L336)

Maybe we tweak that one ^ to be "Failed attempt to evaluate rule"?

Reply:
I think that's a good change to make, though a bit orthogonal to the intent of this PR, which was more about fixing the all-attempt one.

This retry logic is _incredibly_ tangled and written in a confusing way - which is why I didn't really want to touch it here. I had hoped to eventually completely refactor it into what recording rules do, which rewrote the retry implementation from the ground up and is IMO much more understandable.

Reply:
True, at least it works better with this change. We can improve in follow ups.